### PR TITLE
=doc explain compatibility with akka 2.5

### PR DESCRIPTION
**Pull request by ktoso**
Explained in depth a bit about how to use 2.5 with 10.0 as well as linked to bin compat rules.
The question keeps coming up so good to have it in docs

### CODE REVIEW

File: docs/src/main/paradox/java/http/compatibility-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+the rest of your Akka environment, for example like this:
+
+```
+val akkaVersion = "2.5.3"
+val akkaHttpVersion = "$project.version$"
+
+libraryDependencies += "com.typesafe.akka" % "akka-http_2.12" % akkaVersion
+libraryDependencies += "com.typesafe.akka" % "akka-http_2.12" % akkaVersion
+libraryDependencies += "com.typesafe.akka" % "akka-http_2.12" % akkaHttpVersion
+```
```
Comment (jlprat) [inline docs/src/main/paradox/java/http/compatibility-guidelines.md]: What about making this a multilanguage snippet, showing here sbt, gradle and maven dependencies?

File: docs/src/main/paradox/scala/http/compatibility-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+the rest of your Akka environment, for example like this:
+
+```
+val akkaVersion = "2.5.3"
+val akkaHttpVersion = "$project.version$"
+
+libraryDependencies += "com.typesafe.akka" % "akka-http_2.12" % akkaVersion
+libraryDependencies += "com.typesafe.akka" % "akka-http_2.12" % akkaVersion
+libraryDependencies += "com.typesafe.akka" % "akka-http_2.12" % akkaHttpVersion
+```
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/compatibility-guidelines.md]: What about adding a Scala version variable, so people won't think this solution is only for 2.12?

File: docs/src/main/paradox/java/http/compatibility-guidelines.md (hunk 1 above)
Comment (ktoso) [inline docs/src/main/paradox/java/http/compatibility-guidelines.md]: Good point, I'll do so

File: docs/src/main/paradox/scala/http/compatibility-guidelines.md (hunk 2 above)
Comment (ktoso) [inline docs/src/main/paradox/scala/http/compatibility-guidelines.md]: Okey, will do 👍

Comment (ktoso) [inline docs/src/main/paradox/scala/http/compatibility-guidelines.md]: That will still then just show one version, but yeah I'll use the variable - good point :)

Comment (ktoso) [inline docs/src/main/paradox/scala/http/compatibility-guidelines.md]: That will still then just show one version, but yeah I'll use the variable - good point :)

File: docs/src/main/paradox/java/http/compatibility-guidelines.md [hunk 3]
```diff
@@ -30,13 +30,47 @@ dependency MUST be the same version as `akka-actor` (so the exact version from t
...
+    ```
+    val akkaVersion = "2.5.[...]"
+    val akkaHttpVersion = "$project.version$"
+    
+    libraryDependencies += "com.typesafe.akka" %% "akka-http"   % akkaHttpVersion
+    
+    libraryDependencies += "com.typesafe.akka" %% "akka-actor"  % akkaVersion
+    libraryDependencies += "com.typesafe.akka" %% "akka-stream" % akkaVersion
+    ```
+    @@@
```
Comment (ktoso) [inline docs/src/main/paradox/java/http/compatibility-guidelines.md]: This does not render properly for me... I was staring at it quite some time and can't see where I made a typo in the syntax, help?

Comment (ktoso) [inline docs/src/main/paradox/java/http/compatibility-guidelines.md]: Neither does it render the actual substitute project version... but the syntax is correct I believe, we have it in a different place the same style hm...

**Comment by ktoso** [comment]
Fixed :) The trouble with the tabs was very weird... Now works though

File: docs/src/main/paradox/java/http/compatibility-guidelines.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    ```
+    val akkaVersion = "2.5.[...]"
+    val akkaHttpVersion = "$project.version$"
+    libraryDependencies += "com.typesafe.akka" %% "akka-http"   % akkaHttpVersion
+    libraryDependencies += "com.typesafe.akka" %% "akka-actor"  % akkaVersion
+    libraryDependencies += "com.typesafe.akka" %% "akka-stream" % akkaVersion
+    ```
+    @@@
+
+Gradle
```
Comment (jlprat) [inline docs/src/main/paradox/java/http/compatibility-guidelines.md]: being super nitpicky now, on http://akka.io/docs/ it's written `sbt`, `gradle` and `maven`, all in lower case.

Comment (ktoso) [inline docs/src/main/paradox/java/http/compatibility-guidelines.md]: lol ;-) Technically I think these writing styles are the correct ones, sbt insists on writing it as `sbt` and the other ones are names so large, just like we do with Java/Scala. So I'd rather keep this, and fix other places :)

Comment (jlprat) [inline docs/src/main/paradox/java/http/compatibility-guidelines.md]: As I said, super nitpicky. I have your same opinion, I've always seen Maven and Gradle written with the first letter capitalized.

**Comment by ktoso** [comment]
Refs https://github.com/akka/akka-http/issues/1106

**Comment by ktoso** [comment]
No difference, will replace with symlink :)

File: docs/src/main/paradox/java/http/compatibility-guidelines.md [hunk 5]
```diff
@@ -0,0 +1 @@
+../../scala/http/compatibility-guidelines.md
```
Comment (ktoso) [inline docs/src/main/paradox/java/http/compatibility-guidelines.md]: is symlink now

**Comment by ktoso** [comment]
Thanks for review, made it a symlink 👍

**Pull request merged**